Lib/test/test_compile.py

```diff
@@ -2648,6 +2649,24 @@ def test_nested(self):
         self.compare_instructions(seq, [('LOAD_CONST', 1, 1, 0, 0, 0)])
         self.compare_instructions(seq.get_nested()[0], [('LOAD_CONST', 2, 2, 0, 0, 0)])
 
+    def test_static_attributes(self):
+        code = (
+            'class T:\n'
+            '    def __init__(self):\n'
+            '        self.{V1} = 10\n'
+            '        self.{V2} = 10\n'
+            '    def foo(self):\n'
+            '        self.{V3} = self.{V4}\n'
+        )
+        for perm in itertools.permutations(["a", "b", "c", "d"]):
+            var_names = { f'V{i + 1}': perm[i] for i in range(len(perm)) }
+            exec_locals = {}
+            exec(code.format(**var_names), {}, exec_locals)
+            t = exec_locals['T']
+            static_attributes_sorted = tuple(sorted(t.__static_attributes__))
+            self.assertEqual(t.__static_attributes__, static_attributes_sorted)
```

Comment:
```suggestion
        attributes = ('a', 'b', 'c', 'd')
        for perm in itertools.permutations(attributes):
            var_names = { f'V{i + 1}': perm[i] for i in range(len(perm)) }
            exec_locals = {}
            exec(code.format(**var_names), {}, exec_locals)
            t = exec_locals['T']
            self.assertEqual(t.__static_attributes__, attributes)
```
Test that the attributes are actually correct.

Reply:
The fun thing is that actually static attributes don't contain one of the attributes. After reverting this patch running the test you provided outputs following:
```
AssertionError: Tuples differ: ('b', 'a', 'c') != ('a', 'b', 'c', 'd')

First differing element 0:
'b'
'a'

Second tuple contains 1 additional elements.
First extra element 3:
'd'

- ('b', 'a', 'c')
+ ('a', 'b', 'c', 'd') : code is class T:
    def __init__(self):
        self.a = 10
        self.b = 10
    def foo(self):
        self.c = self.d
```
So `self.d` is missing. @JelleZijlstra , what should I do about rewriting this test? If I keep it as it was then this can be fixed in the follow up as it is a different issue

Reply:
Oh, I was confused about that too. Since `static_attributes` only contains attributes that are *assigned* to, you should simply rewrite the test so it doesn't add attributes that are only read.